### ISSUE TITLE
Question: Benefit and usage of clip_grad_norm

### ISSUE THREAD

**Issue by shahjaidev** [RUCAIBox/RecBole#1396]
This is more of a knowledge question: what is the role and benefit of the clip_grad_norm setting? When should one use gradient clipping?
That is, can you share any of the model papers implemented in the recommendation space that use this for improving results

Thanks!

**Comment by zhengbw0324** [comment]
As the neural network becomes deeper, the gradient multiplier in the back propagation process will increase, which is more likely to cause gradients vanishing and exploding. For the gradients exploding problem, one of the solutions is gradient clipping, that is, to control the upper limit of the gradient.

`torch.nn.utils.clip_grad_norm_()` is a gradient clipping function in pytorch. For more details, please refer to the [Pytorch documentation](https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html). And you can search for more relevant principles and usage methods on Google.

**Comment by shahjaidev** [comment]
Understood, thanks! Closing the issue
